core/controlplane/config/templates/cloud-config-controller

```diff
@@ -1829,6 +1833,118 @@ write_files:
+                effect: NoSchedule
+              - operator: Exists
+                effect: NoExecute
+              - operator: Exists
+                key: CriticalAddonsOnly
+              volumes:
+              - name: kube-dns-config
+                configMap:
+                  name: kube-dns
+                  optional: true
+              containers:
+              - name: dnsmasq
+                image: {{ .KubeDnsMasqImage.RepoWithTag }}
+                livenessProbe:
+                  httpGet:
+                    path: /healthcheck/dnsmasq
+                    port: 10054
+                    scheme: HTTP
+                  initialDelaySeconds: 60
+                  timeoutSeconds: 5
+                  successThreshold: 1
+                  failureThreshold: 5
+                args:
+                - -v=2
+                - -logtostderr
+                - -configDir=/etc/k8s/dns/dnsmasq-nanny
+                - -restartDnsmasq=true
+                - --
+                - -k
+                - --cache-size=1000
+                - --server=/cluster.local/{{.DNSServiceIP}}
+                - --server=/in-addr.arpa/{{.DNSServiceIP}}
+                - --server=/ip6.arpa/{{.DNSServiceIP}}
+                - --log-facility=-
+                ports:
+                - containerPort: 53
+                  name: dns
+                  protocol: UDP
+                - containerPort: 53
+                  name: dns-tcp
+                  protocol: TCP
+                # see: https://github.com/kubernetes/kubernetes/issues/29055 for details
+                resources:
+                  requests:
+                    cpu: 150m
+                    memory: 20Mi
+                volumeMounts:
+                - name: kube-dns-config
+                  mountPath: /etc/k8s/dns/dnsmasq-nanny
+              - name: sidecar
+                image: {{ .DnsMasqMetricsImage.RepoWithTag }}
+                livenessProbe:
+                  httpGet:
+                    path: /metrics
+                    port: 10054
+                    scheme: HTTP
+                  initialDelaySeconds: 60
+                  timeoutSeconds: 5
+                  successThreshold: 1
+                  failureThreshold: 5
+                args:
+                - --v=2
+                - --logtostderr
+                - --probe=dnsmasq,127.0.0.1:53,ec2.amazonaws.com,5,A
+                ports:
+                - containerPort: 10054
+                  name: metrics
+                  protocol: TCP
+                resources:
+                  requests:
+                    memory: 20Mi
+              - name: healthz
+                image: {{ .ExecHealthzImage.RepoWithTag }}
+                resources:
+                  limits:
+                    memory: 50Mi
+                  requests:
+                    cpu: 10m
+                    memory: 50Mi
+                args:
+                - --cmd=nslookup ec2.amazonaws.com 127.0.0.1 >/dev/null
```

Comment:
`ec2.amazonaws.com` is resolvable even from your laptop, so it should be fine in any setup.

I just found that in kube-aws kube-dns runs with empty healthz container, which is completely pointless. I'll make healthz to look exactly the same as it is now in kube-aws , then' we'll address what to do with healthz in a separate PR